calculator.js

```diff
@@ -1,5 +1,86 @@
-const exampleAdditionInput = {
+const exampleInput = {
   num1: 3,
-  num2: 5,
-  operation: 'add',
+  num2: 2,
+  operation: "-"
+};
+
+// VALID OPERATORS
+const validValues = [
+  "add",
+  "+",
+  "subtract",
+  "-",
+  "multiply",
+  "*",
+  "divide",
+  "/",
+  "%",
+  "exponify",
+  "**"
+];
+
+if (!validValues.includes(exampleInput.operation)) {
+  console.log("Operation is not valid. Please check your operation value");
 }
+
+const isZeroNum = function(num) {
+  return num === 0;
+};
+
+const isNumber = function(value) {
+  return typeof value === "number" && isFinite(value); //checking data types: https://webbjocke.com/javascript-check-data-types/
+};
+
+const runOperation = function(numOne, numTwo, operation) {
+  let operator;
+  let result;
+  switch (operation) {
+    case "add":
+    case "+":
+      result = numOne + numTwo;
+      operator = "+";
+      break;
+
+    case "subtract":
+    case "-":
+      result = numOne - numTwo;
+      operator = "-";
+      break;
+
+    case "multiply":
+    case "*":
+      result = numOne * numTwo;
+      operator = "*";
+      break;
+
+    case "divide":
+    case "/":
+    case "%": {
+      if (isZeroNum(numTwo)) {
+        console.log("Second number should be greater than 0 for division.");
```

Comment:
This error message isn't quite right. Dividing by a negative number is OK (and your program can handle it), it's just 0 that's a problem

Reply:
Thank you so much!